File: mambo-protocol/src/main/scala/org/mambo/protocol/messages/game/inventory/exchanges/ExchangeStartedWithStorageMessage.scala





















// Generated on 05/03/2013 10:34:56
// Scala version by Blackrush
package org.mambo.protocol.messages

import org.mambo.core.io._
import org.mambo.protocol.{Message => BaseMessage, _}
import org.mambo.protocol.types._
import org.mambo.protocol.enums._

trait ExchangeStartedWithStorageMessage extends ExchangeStartedMessage {
  override val messageId = 6236

  protected trait Message extends ExchangeStartedMessage.Message with Serializable {
    override val messageId = ExchangeStartedWithStorageMessage.messageId

    val storageMaxSlot: Int

    override def serialize(buf: Buffer) {
      super.serialize(buf)

      buf.writeInt(storageMaxSlot)
    }

  }

  protected def create(exchangeType: Byte, storageMaxSlot: Int): Target = throw new IllegalAccessException

  override def deserialize(buf: Buffer) = {
    val exchangeType = buf.readByte()
    val storageMaxSlot = buf.readInt()

    create(exchangeType, storageMaxSlot)
  }

}

object ExchangeStartedWithStorageMessage extends ExchangeStartedWithStorageMessage {
  case class MessageImpl(exchangeType: Byte, storageMaxSlot: Int) extends Message
  type Target = MessageImpl

  override def create(exchangeType: Byte, storageMaxSlot: Int) = MessageImpl(exchangeType, storageMaxSlot)

  def apply(exchangeType: Byte, storageMaxSlot: Int) = create(exchangeType, storageMaxSlot)
  def unapply(msg: MessageImpl) = MessageImpl.unapply(msg)
}

File: mambo-protocol/src/main/java/org/mambo/protocol/messages/ExchangeStartedWithStorageMessage.java


// Generated on 05/08/2013 19:37:55
package org.mambo.protocol.messages;

import java.util.*;
import org.mambo.protocol.types.*;
import org.mambo.protocol.enums.*;
import org.mambo.protocol.*;
import org.mambo.core.io.*;

public class ExchangeStartedWithStorageMessage extends ExchangeStartedMessage {
    public static final int MESSAGE_ID = 6236;
    
    @Override
    public int getNetworkMessageId() {
        return MESSAGE_ID;
    }
    
    public int storageMaxSlot;
    
    public ExchangeStartedWithStorageMessage() { }
    
    public ExchangeStartedWithStorageMessage(byte exchangeType, int storageMaxSlot) {
        super(exchangeType);
        this.storageMaxSlot = storageMaxSlot;
    }
    
    @Override
    public void serialize(Buffer buf) {
        super.serialize(buf);
        buf.writeInt(storageMaxSlot);
    }
    
    @Override
    public void deserialize(Buffer buf) {
        super.deserialize(buf);
        storageMaxSlot = buf.readInt();
        if (storageMaxSlot < 0)
            throw new RuntimeException("Forbidden value on storageMaxSlot = " + storageMaxSlot + ", it doesn't respect the following condition : storageMaxSlot < 0");
    }
    
}
